lib/server-jobs.js

```diff
@@ -492,15 +498,27 @@ module.exports.getJobSequenceAsync = util.promisify(module.exports.getJobSequenc
 module.exports.getJobSequenceWithFormattedOutput = function (job_sequence_id, course_id, callback) {
   module.exports.getJobSequence(job_sequence_id, course_id, (err, jobSequence) => {
     if (ERR(err, callback)) return;
-    (jobSequence.jobs || []).forEach((job) => {
+
+    // Generate a token to authorize websocket requests for this job sequence.
+    const jobSequenceTokenData = {
+      jobSequenceId: job_sequence_id.toString(),
+    };
+    jobSequence.token = csrf.generateToken(jobSequenceTokenData, config.secretKey);
+
+    (jobSequence.jobs ?? []).forEach((job) => {
+      // Also generate a token for each job.
+      const jobTokenData = { jobId: job.id.toString() };
+      job.token = csrf.generateToken(jobTokenData, config.secretKey);
+
```

Comment:
Why are we doing this here and not inside `getJobSequence()`?

Reply:
Great question! Because I started at the pages that render the live output and walked back to this callsite and no further 😂 I'll move it to that other function.

Reply:
Done in cc1e319.